lib/Doctrine/MongoDB/Aggregation/Expr.php

```diff
@@ -584,6 +584,103 @@ public function ifNull($expression, $replacementExpression)
     }
 
     /**
+     * Returns a boolean indicating whether a specified value is in an array.
+     *
+     * Unlike the $in query operator, the aggregation $in operator does not
+     * support matching by regular expressions.
+     *
+     * @see https://docs.mongodb.com/manual/reference/operator/aggregation/in/
+     * @since 1.5
+     * @param mixed|self $expression
+     * @param mixed|self $arrayExpression
+     * @return Expr
+     */
+    public function in($expression, $arrayExpression)
+    {
+        return $this->operator('$in', [$expression, $arrayExpression]);
+    }
+
+    /**
+     * Searches an array for an occurence of a specified value and returns the
+     * array index (zero-based) of the first occurence. If the value is not
+     * found, returns -1.
+     *
+     * @see https://docs.mongodb.com/manual/reference/operator/aggregation/indexOfArray/
+     * @since 1.5
+     * @param mixed|self $arrayExpression Can be any valid expression as long as it resolves to an array.
+     * @param mixed|self $searchExpression Can be any valid expression.
+     * @param mixed|self $start Optional. An integer, or a number that can be represented as integers (such as 2.0), that specifies the starting index position for the search. Can be any valid expression that resolves to a non-negative integral number.
+     * @param mixed|self $end An integer, or a number that can be represented as integers (such as 2.0), that specifies the ending index position for the search. Can be any valid expression that resolves to a non-negative integral number.
+     * @return Expr
+     */
+    public function indexOfArray($arrayExpression, $searchExpression, $start = null, $end = null)
+    {
+        $args = [$arrayExpression, $searchExpression];
+        if ($start !== null) {
+            $args[] = $start;
+        }
+        if ($end !== null) {
+            $args[] = $end;
+        }
```

Comment:
Should it be a logical error if `$start` is null but `$end` is non-null? Otherwise, `$end` will be appended to `$args` in the position where `$start` is expected.